src/patternfly/components/Page/page.scss

```diff
@@ -272,50 +279,51 @@
   margin-bottom: var(--pf-c-page__header-tools--MarginBottom);
   margin-left: auto; // to push it to the right
 
-  .pf-m-icons {
+  // Hides on mobile visible on desktop
+  .pf-m-desktop {
     display: none;
     visibility: hidden;
 
     @media screen and (min-width: $pf-global--breakpoint--lg) {
-      display: block;
+      display: inline-block;
       visibility: visible;
     }
   }
 
+  // Hides on desktop visible on mobile
   .pf-m-mobile {
-    margin-right: 0;
+    display: inline-block;
+    visibility: visible;
 
     @media screen and (min-width: $pf-global--breakpoint--lg) {
       display: none;
       visibility: hidden;
     }
   }
 
-  .pf-m-user {
+  .pf-c-avatar {
     display: none;
+    margin-left: 0;
     visibility: hidden;
 
-    @media screen and (min-width: $pf-global--breakpoint--md) {
+    @media screen and (min-width: $pf-global--breakpoint--lg) {
+      margin-left: var(--pf-c-page__header-tools--c-avatar--MarginLeft);
+    }
+
+    @media screen and (min-width: $pf-global--breakpoint--sm) {
       display: block;
       visibility: visible;
     }
   }
 
-  .pf-c-avatar {
-    margin-left: var(--pf-c-page__header-tools--c-avatar--MarginLeft);
-  }
-
   @media screen and (min-width: $pf-global--breakpoint--lg) {
     grid-column: 3 / 4;
   }
 }
 
 .pf-c-page__header-tools-group {
   display: flex;
-
-  & + & {
-    margin-left: var(--pf-c-page__header-tools-group--MarginLeft);
-  }
+  margin-left: var(--pf-c-page__header-tools-group--MarginLeft);
```

Comment:
With this change, this is no longer true

> there is only a margin-left on the .pf-c-page__header-tools-group when its next to .pf-c-page__header-tools-group .pf-m-icons

Is that expected? Personally I think it's good. I might even increase the margin-left on mobile, since the logo can span as wide as the available space in the header now, meaning you can end up with only 8px space between the utility icons and the header logo if the logo is wide enough

<img width="355" alt="Screen Shot 2019-07-02 at 1 38 37 PM" src="https://user-images.githubusercontent.com/35148959/60537725-bf687200-9cce-11e9-8141-2897126ca0ed.png">